### PR TITLE
/cars api endoint

### LINKED ISSUE #41: handle filter api /cars?
- add endpoint /cars.
- add filter query.
- write tests to check the endpoint.
- ...

### PR DESCRIPTION
**Pull request by mai-mohammed**
In this branch I do the following:

- add controllers wrapper function. 
- install required test packages ```[@types/jest, @types/supertest]```.
- edit test script by appending ```cross-env NODE_ENV=development```.
- handle the endpoint (route, controller, query).
- write tests. (please suggest more tests).

### CODE REVIEW

**Review by tariqabusada — APPROVED** [review]

**Review by husamkamal — APPROVED** [review]

**Review by abdou059 — APPROVED** [review]

File: server/src/routes/routes.ts [hunk 1]
```diff
@@ -1,9 +1,13 @@
...
+import ExpressWrapper from './ExpressWrapper';
+import getFilteredCars from '../controllers/cars/getFilteredCars';
 
 const router = Router();
 
 router.get('/', (req: Request, res: Response) => {
   res.send('hello');
 });
 
+router.get('/cars', ExpressWrapper(getFilteredCars));
```
Comment (LinaYahya) [inline server/src/routes/routes.ts]: change file name to index.ts and here try to use all routes we have,
So create SubRoute for cars and here you can use it

File: server/src/routes/ExpressWrapper.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+// const createError = require('http-errors');
```
Comment (LinaYahya) [inline server/src/routes/ExpressWrapper.ts]:
```suggestion
```

File: server/src/db/queries/carsQueries/getCarsQuery.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import { Op } from 'sequelize';
+import { Car } from '../../models';
+
+const getCarsQuery = async ({
```
Comment (LinaYahya) [inline server/src/db/queries/carsQueries/getCarsQuery.ts]: let us add `state` and `page` as search params, and return matching results count using `findAndCountAll`,
Also I think No need to trim values here,
And maybe we can validate `req.query`

File: server/src/db/queries/carsQueries/getCarsQuery.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  brand, model, year, maxPrice, fuel, mileage, goodPrice,
+}) => {
+  const where:any = {
+  };
+  if (brand !== null && brand !== '') {
```
Comment (LinaYahya) [inline server/src/db/queries/carsQueries/getCarsQuery.ts]:
```suggestion
  if (brand) { // the same for other fields
```

**Review by LinaYahya — CHANGES_REQUESTED** [review]

**Review by LinaYahya — APPROVED** [review]

**Pull request merged**